src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -391,6 +399,60 @@ public static TimeSpan MaximumTimeout
             }
         }
 
+        /// <summary>
+        /// Controls whether usage of System.Transactions APIs that require escalation to a distributed transaction will do so;
+        /// if your application requires distributed transaction, opt into using them by setting this to <see langword="true" />.
+        /// If set to <see langword="false" /> (the default), escalation to a distributed transaction will throw a <see cref="NotSupportedException" />.
+        /// </summary>
+#if WINDOWS
+        public static bool ImplicitDistributedTransactions
+        {
+            get => DtcProxyShimFactory.s_transactionConnector is not null;
+
+            [SupportedOSPlatform("windows")]
+            [RequiresUnreferencedCode(DistributedTransactionTrimmingWarning)]
+            set
+            {
+                // Make sure this flag can only be set once, and that once distributed transactions have been initialized,
+                // it's frozen.
+                if (ImplicitDistributedTransactionsInternal is null)
+                {
+                    ImplicitDistributedTransactionsInternal = value;
+                }
+                else if (value != ImplicitDistributedTransactionsInternal)
+                {
+                    throw new InvalidOperationException(SR.ImplicitDistributedTransactionsCannotBeChanged);
+                }
+
+                if (value)
+                {
+                    DtcProxyShimFactory.s_transactionConnector ??= new DtcProxyShimFactory.DtcTransactionConnector();
+                }
+                else
+                {
+                    DtcProxyShimFactory.s_transactionConnector = null;
+                }
+            }
+        }
+
+        internal static bool? ImplicitDistributedTransactionsInternal;
```

Comment:
```suggestion
        internal static bool? s_implicitDistributedTransactions;
```
The repo naming convention for statis fields is to start with `s_`. Once you have the `s_` prefix, you do not need `Internal` to disambiguate.